### PR TITLE
chore : migrated to github actions and removed travis ci

### LINKED ISSUE #192: Migrate to github actions
**Is your feature request related to a problem? Please describe.**
Add github actions to test pull requests so we can get a precise report of the pull request whether the tests are passing or not etc.

### PR DESCRIPTION
**Pull request by vj-codes**
### Description
Firstly removed travis ci and added github actions that do the same task as travis 

Fixes #192 

### Type of Change:
- Code
- Quality Assurance

### How Has This Been Tested?
The build passes on my fork here https://github.com/vj-codes/bridge-in-tech-backend/actions

### Checklist:

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by epicadk — DISMISSED** [review]
💯

File: .github/workflows/main.yml
Comment (mtreacy002) [inline .github/workflows/main.yml]: @vj-codes , how come the 'develop' branch is not specify here? atm this PR is not triggering a gh-action build. Perhaps this has something to do with it?

File: .github/workflows/main.yml
Comment (mtreacy002) [inline .github/workflows/main.yml]: here we only need DB_TEST_NAME. DB_NAME is not being used coz gh-action only run the tests.

File: .github/workflows/main.yml
Comment (mtreacy002) [inline .github/workflows/main.yml]: is this duplication from line 27? I don't think we need it again here, do we?

File: .github/workflows/main.yml
Comment (mtreacy002) [inline .github/workflows/main.yml:58]: I think we can just test coverage on tests folder, not the whole project, so this line can be removed.

File: .github/workflows/main.yml
Comment (mtreacy002) [inline .github/workflows/main.yml:60]: Why is the report didn't get uploaded to Codecov?

**Review by mtreacy002 — CHANGES_REQUESTED** [review]
@vj-codes . Thank you for submitting the PR. I have few questions/suggestions below.

File: .github/workflows/main.yml
Comment (vj-codes) [reply .github/workflows/main.yml]: the current code will trigger gh actions every time a contributor pushes the changes on their fork as well as every time a pull request is created here irrespective of any branch .
The build is passing under actions on my fork and will be triggered by other PRs here after this is merged.

File: .github/workflows/main.yml
Comment (vj-codes) [reply .github/workflows/main.yml]: ahh okay will update it

File: .github/workflows/main.yml
Comment (vj-codes) [reply .github/workflows/main.yml:58]: the tests folder was recently removed from coverage on ms-backend so I did the same here , should I stick to tests only?

File: .github/workflows/main.yml
Comment (vj-codes) [reply .github/workflows/main.yml:60]: because there's no codecov.yml or codecov bot here so I left it as a comment and could be updated when that codecov PR is merged

File: .github/workflows/main.yml
Comment (mtreacy002) [reply .github/workflows/main.yml]: @vj-codes , I believe if we speccify the `develop` as the target branch for pull and push, it will trigger the gh-action here on this PR even before we merge it. This way we can see if the build pass or not before merging the code, which is much better than if we wait till the PR is merged and wait for another PR to be pushed to check if it works or not. You can see this example on my MS PR [# 1029](https://github.com/anitab-org/mentorship-backend/pull/1029). Initially I left the branch as develop as that's what the existing code has (as `bit` branch was branched out of the `develop` branch). However, since it didn't trigger the build, I changed it to `bit` which is the target branch I want to merge my PR into. Only then it triggered the build there (`bit` branch), even though the PR is still not yet merged. 
IMO for this PR, it is more important to make sure that the build is passing on the actual `develop` branch before the code being merge rather than to merge first then triggers the build later with the next PR. What do you think?

File: .github/workflows/main.yml
Comment (mtreacy002) [reply .github/workflows/main.yml:58]: IMO no need to check the coverage for the whole project, just the tests folder. But perhaps @isabelcosta can give us feedback on this.

Comment (epicadk) [reply .github/workflows/main.yml:58]: > yes, please. No need to check the coverage for the whole project, just the tests folder.

I think you mean the tests don't need to be included in the coverage check 😅.

Comment (mtreacy002) [reply .github/workflows/main.yml:58]: > > yes, please. No need to check the coverage for the whole project, just the tests folder.
> 
> I think you mean the tests don't need to be included in the coverage check 😅.

oh... oops... my bad. Sorry, @vj-codes, yep, you can keep it as --cov=. then 😉

**Review by epicadk — DISMISSED** [review]
👍🏼

**Comment by vj-codes** [comment]
> 👍🏼

@epicadk the build is still not coming here in the PR and didn't show up on my fork as well after the changes were made but @mtreacy002 says it should show up here

File: .github/workflows/main.yml
Comment (mtreacy002) [reply .github/workflows/main.yml]: @epicadk , @vj-codes , do we need this line (43)

**Comment by epicadk** [comment]
> > 👍🏼
> 
> @epicadk the build is still not coming here in the PR and didn't show up on my fork as well after the changes were made but @mtreacy002 says it should show up here

Is github actions enabled for the repo?

**Comment by epicadk** [comment]
Also I remember something similar happening in mentorship android maybe @isabelcosta can help us out.

File: .github/workflows/main.yml
Comment (codesankalp) [reply .github/workflows/main.yml]: Yes, it is not needed again.

**Comment by vj-codes** [comment]
@mtreacy002 I have made the changes still the build is not triggered , is this and black merged to develop before OSH?
let me know if any changes are required 😉

File: .github/workflows/main.yml
Comment (codesankalp) [inline .github/workflows/main.yml]:
```suggestion
    branches: [develop]
```

File: .github/workflows/main.yml
Comment (codesankalp) [inline .github/workflows/main.yml]:
```suggestion
    branches: [develop]
```

**Review by codesankalp — CHANGES_REQUESTED** [review]
I think after making suggested changes it will trigger the build @vj-codes

**Comment by vj-codes** [comment]
@codesankalp it didn't

**Comment by codesankalp** [comment]
@vj-codes It will work.
Also, remove whitespaces from pull-request one.
See this: https://github.com/codesankalp/bridge-in-tech-backend/actions/runs/698795020
Or, just replace your main.yml with this: https://github.com/codesankalp/bridge-in-tech-backend/blob/ghBIT/.github/workflows/main.yml

**Comment by vj-codes** [comment]
@codesankalp it did work on my repo, thanks!
Although I don't think the spaces make any difference, what @mtreacy002 meant was that the build should trigger here on the main repo even before this PR is merged which isn't happening. Do you know if it's possible or not?

**Comment by mtreacy002** [comment]
> @mtreacy002 I have made the changes still the build is not triggered , is this and black merged to develop before OSH?
> let me know if any changes are required 😉

@vj-codes, it would be great if we could merge these 2 before OSH. I'm waiting for @isabelcosta to respond to [my zulip post](https://anitab-org.zulipchat.com/#narrow/stream/237630-bridge-in-tech/topic/help/near/232208275) to whether or not she has enabled the gh-actions setting from her side. Perhaps @annabauza can also help us on this?

File: .github/workflows/main.yml
Comment (epicadk) [reply .github/workflows/main.yml]: The whitespace won't matter. You can check the mentorship backend workflow there is whitespace there as well.

**Comment by vj-codes** [comment]
> @vj-codes, it would be great if we could merge these 2 before OSH. I'm waiting for @isabelcosta to respond to [my zulip post](https://anitab-org.zulipchat.com/#narrow/stream/237630-bridge-in-tech/topic/help/near/232208275) to whether or not she has enabled the gh-actions setting from her side. Perhaps @annabauza can also help us on this?
>
cc : @anitab-org/admins

**Review by isabelcosta — APPROVED** [review]
seems good :) Happy to accept this. Will leave last decision to @mtreacy002

**Review by mtreacy002 — APPROVED** [review]
LGTM, @vj-codes . Will approve this now 🎉. Thank you again for your contribution 🎉🎉

**Comment by isabelcosta** [comment]
@vj-codes Awesome work 👏🏾 🎉 🙌🏾

**Pull request merged**
